### PR TITLE
Initial working module

### PR DESCRIPTION
**Pull request by mbentz-uf**

### CODE REVIEW

File: js/pastFutureDateTags.js [hunk 1]
```diff
@@ -0,0 +1,347 @@
...
+		fn(field);
+	}
+}
+
+// Sets the maxDate option on the jQuery UI datepicker
+function preventFutureDate(field) {
+	let $input = $(`#${field}-tr input`);
+	let val = $input.attr("onblur");
+	let dateFormat = $input.attr('fv');
+	let [minDate, _] = val.match(reDates);
```
Comment (ChemiKyle) [inline js/pastFutureDateTags.js:73]: `Uncaught TypeError: can't access property Symbol.iterator, val.match(...) is null`

**Comment by ChemiKyle** [comment]
If a field already has data in it, the module will automatically "correct" it with no fanfare, before the date selector is opened.


https://user-images.githubusercontent.com/20332546/105391475-28347980-5be8-11eb-8a71-d97ffa832031.mov



I'm not sure we want that by default since it alters data unexpectedly. This might need to go behind a project level setting.

**Comment by mbentz-uf** [comment]
I am not sure of the best way of handling this scenario. The behavior of silently correcting the date is caused by setting a min/max date on a datepicker via `$('.selector').datepicker("option", "minDate", "-0d");`. Correcting the behavior is not without side effects:

#### Setup
```
$input // input associated with datepicker

Date field was initially saved on: 01/22/2021
$input.val(): 01/22/2021  // onblur event on this field triggers validation 
$input.attr('val'): 01/22/2021 // populated after instrument is saved

Date form is revisited on: 01/23/2021
$input.val() (on load): 1/23/2021 // valid date based on minDate restriction
$input.attr('val') (on load): 1/22/2021
```
#### Assumption
Date fields tagged with __@PREVENT-FUTUREDATE__ or __@PREVENT-PASTDATE__ and data in the `value` attribute were set correctly before.

#### Fix
Update `input.val()` with the value in `input.attr('val')`, and prevent validation until the user attempts to make another change. 

#### Side effects
Given the scenario above, an infinite loop of failed validation will occur for that field that was set correctly before with the value of __1/22/2021__, but now fails validation when the form is revisited on __1/23/2021__ because `input.val()` is set to __1/22/2021__ (a past date) when the `onblur` event is triggered. 

***
This is complicated to fix because there is no guarantee that a date was set correctly before and also causes the scenario described in the README:

>Screen Shot 2021-01-22 at 10 12 10 AM

**Comment by mbentz-uf** [comment]
> I am not sure of the best way of handling this scenario. The behavior of silently correcting the date is caused by setting a min/max date on a datepicker via `$('.selector').datepicker("option", "minDate", "-0d");`. Correcting the behavior is not without side effects. See scenario below:
> 
> #### Setup
> ```
> $input // input associated with datepicker
> 
> Date field was initially saved on: 01/22/2021
> $input.val(): 01/22/2021  // onblur event on this field triggers validation 
> $input.attr('val'): 01/22/2021 // populated after instrument is saved
> 
> Date form is revisited on: 01/23/2021
> $input.val() (on load): 1/23/2021 // valid date based on minDate restriction
> $input.attr('val') (on load): 1/22/2021
> ```
> 
> #### Assumption
> Date fields tagged with **@PREVENT-FUTUREDATE** or **@PREVENT-PASTDATE** and data in the `value` attribute were set correctly before.
> 
> #### Fix
> Update `input.val()` with the value in `input.attr('val')`, and prevent validation until the user attempts to make another change.
> 
> #### Side effects
> Given the scenario above, an infinite loop of failed validation will occur for that field that was set correctly before with the value of **1/22/2021**, but now fails validation when the form is revisited on **1/23/2021** because `input.val()` is set to **1/22/2021** (a past date) when the `onblur` event is triggered.
> 
> This is complicated to fix because there is no guarantee that a date was set correctly before and also causes the scenario described in the README:
> 
> > Screen Shot 2021-01-22 at 10 12 10 AM

@ChemiKyle It can be left as is or fixed with __above__. A fix without side effects would require adding tables to mysql to validate if fields marked with __@PREVENT-FUTUREDATE/@PREVENT-PASTDATE__ passed validation the last time they were updated and apply/bypass validation based on that.

**Comment by ChemiKyle** [comment]
Update works, data is no longer "corrected" in-place. I'm going to make a small update to the README and merge.

**Pull request merged**